src/game/logic/ai/aipathfind.cpp

```diff
@@ -16,6 +16,220 @@
 #ifdef GAME_DLL
 #include "hooker.h"
 #endif
+#include "xfer.h"
+
+PathfindZoneManager::PathfindZoneManager() :
+    m_zoneBlocks(nullptr),
+    m_zoneBlockPointers(nullptr),
+    m_maxZone(0),
+    m_updateFrequency(0),
+    m_zoneTableSize(0),
+    m_zoneTable1(nullptr),
+    m_zoneTable2(nullptr),
+    m_zoneTable3(nullptr),
+    m_zoneTable4(nullptr),
+    m_zoneTable5(nullptr),
+    m_zoneTable6(nullptr)
+{
+    m_blockSize.x = 0;
+    m_blockSize.y = 0;
+}
+
+PathfindZoneManager::~PathfindZoneManager()
+{
+#ifdef GAME_DLL
+    Call_Method<void, PathfindZoneManager>(PICK_ADDRESS(0x0055C320, 0x0088BF4E), this);
+#endif
+}
+
+PathfindLayer::PathfindLayer() : m_map(nullptr), m_mapPointers(nullptr), m_unk(0), m_bridge(nullptr), m_destroyed(false)
+{
+    m_extent.lo.x = 0;
+    m_extent.lo.y = 0;
+    m_extent.hi.x = 0;
+    m_extent.hi.y = 0;
+    m_unkRegion.lo.x = -1;
+    m_unkRegion.lo.y = -1;
+    m_unkRegion.hi.x = -1;
+    m_unkRegion.hi.y = -1;
+}
+
+PathfindLayer::~PathfindLayer()
+{
+    Reset();
+}
+
+void PathfindLayer::Reset()
+{
+#ifdef GAME_DLL
+    Call_Method<void, PathfindLayer>(PICK_ADDRESS(0x0055DCA0, 0x0088E85F), this);
+#endif
+}
+
+#ifndef GAME_DLL
+PathfindCellInfo *PathfindCellInfo::m_infoArray;
+PathfindCellInfo *PathfindCellInfo::m_firstFree;
+#endif
+
+void PathfindCellInfo::Allocate_Cell_Infos()
+{
+    Release_Cell_Infos();
+    m_infoArray = new PathfindCellInfo[30000];
+    m_infoArray[29999].m_pathParent = 0;

```

Comment:
Add a global `constexpr int max_cell_count = 30000` in this file and then use it in these functions, like `max_cell_count - 1` here.